### ISSUE TITLE
Fixed inertial frame mask

### ISSUE THREAD

**Issue by lgrcia** [lgrcia/spotter#44]
Here is how I would go about making a mask on the star that is fixed in the inertial frame.

Let's first define a rotating star with a small bright spot.
```python
from spotter import Star, show
import numpy as np

star = Star.from_sides(120, period=1.0)
star = star.set(y= star.spot(0, 0, 0.05, 300))
```

Here is an example of a mask:

```python
import healpy as hp
import jax.numpy as jnp
import matplotlib.pyplot as plt


def mask_function(star):
    phi, theta = hp.vec2ang(star.x)

    def impl(phase):
        _theta = (theta - phase) % (jnp.pi)
        y = jnp.sin(_theta) * jnp.sin(phi)
        return jnp.where(jnp.abs(jnp.sin(y * 10)) < 0.9, 0, 1)

    return impl


mask = mask_function(star)

plt.subplot(121)
show(star)
plt.subplot(122)
show(Star(y=mask(0.0)))
```

> [!IMPORTANT]
> We might think that the mask should be fixed while the star rotates. But remember that it is [the other way around in spotter](https://spotter.readthedocs.io/en/latest/notebooks/principle/). Hence, the mask has to be transformed to follow spotter's design matrix:

<img width="510" height="240" alt="Image" src="https://github.com/user-attachments/assets/14e0e1e6-dfa0-4a20-903a-28a12e6b546d" />

The plot above shows the spotted star on the left and the mask on the right.

Finally, here is the light curve of the masked star while it rotates:

```python
from spotter.light_curves import design_matrix
import jax.numpy as jnp


def masked_light_curve(star, time, mask, normalize=True):

    def impl(star, time):
        return jnp.einsum(
            "ij,ij->i", design_matrix(star, time) * mask(star.phase(time)), star.y
        )

    norm = 1 / jnp.mean(star.y) if normalize else 1.0

    return jnp.vectorize(impl, excluded=(0,), signature="()->(n)")(star, time).T * norm


time = np.linspace(-0.25, 0.25, 1000)
flux = masked_light_curve(star, time, mask, False)[0]

plt.plot(time, flux)
```

<img width="566" height="405" alt="Image" src="https://github.com/user-attachments/assets/de43a6fa-3ca7-4b7d-b877-413d76a69e9f" />

Few things to consider here:
- the mask has to be recomputed at every phase.
- the mask must have an analytical expression and follow the star's rotation. I did that by hand here but one has to figure how to analytically compute a mask that rotates with an inclined star. The alternative to have a fixed mask and rotate it using some kind of interpolation is possible (using spherical harmonics). My tests suggest that the latter leads to very high errors though but I'd be happy to share the code.

As a reference, when I say the mask has to "rotate", here is what I mean:

![Image](https://github.com/user-attachments/assets/1658cfb6-4ab1-4e34-8080-f6823eeba639)

The left is an orthographic view as seen by the observer, the right is a mollview highlighting that the stellar map is fixed and the filter "rotate" with spotter's design matrix. Again I highly recommend reading [https://spotter.readthedocs.io/en/latest/notebooks/principle/](https://spotter.readthedocs.io/en/latest/notebooks/principle/) to understand why it is so.

An alternative would be to use starry, and express the mask analytically by expanding it in the spherical harmonics basis.

**Comment by frescigno** [comment]
That makes perfect sense!

By following the method described above I have been able to set up what looks like a "fixed" telescope mask. The key was exactly this line of code
`_theta = (theta - phase) % (jnp.pi)`

Once I have included that line my mask is now stable. I have attached a figure (done following the directions in the readthedocs tutorials)

<img width="789" height="240" alt="Image" src="https://github.com/user-attachments/assets/84b09e98-4e40-43cd-9d70-9a44c1f2d39a" />
